src/suricata.c

```diff
@@ -398,6 +398,8 @@ static void GlobalsDestroy(SCInstance *suri)
     MpmHSGlobalCleanup();
 #endif
 
+    /* The pid file name may be in config memory, but is needed later. */
+    suri->pid_filename = strdup(suri->pid_filename);
```

Comment:
Note that `pid_filename` may be `NULL` unless

- Suricata is running as a daemon
- Suricata was started `--pidfile`.

See https://github.com/OISF/suricata/blob/master/src/suricata.c#L2127

Reply:
Right, I have added a != NULL check.